istioctl/cmd/root.go

```diff
@@ -159,8 +159,12 @@ debug and diagnose their Istio mesh.
 	}
 
 	cmd.AddFlags(rootCmd)
+	deprecatedMsg := "it will be removed in Istio 1.9."
 
-	rootCmd.AddCommand(register())
+	registerCmd := register()
+	registerCmd.Deprecated = deprecatedMsg
+	rootCmd.AddCommand(registerCmd)
+	deregisterCmd.Deprecated = deprecatedMsg
```

Comment:
Setting `Deprecated` hides the command.  For 1.8 we should not hide it, so that users doing `istioctl -h` can see a message.  Just `+` a suffix to the `Short`.